### ISSUE TITLE
Do not display LSP indicators for current buffer

### ISSUE THREAD

**Issue by lcrockett** [akinsho/nvim-bufferline.lua#104]
I'm trying to figure out how to write a conditional within `diagnostic_indicator` to only display LSP indicators for visible buffers and have them not appear on the bufferline for the current buffer itself. The rationale behind this is that my statusline and current buffer itself already display sufficient LSP indicators to draw the required attention.

I've tried playing around with a few functions i've found [here](https://github.com/akinsho/nvim-bufferline.lua/blob/4c86c1fa30a6e27a08f1026ebbeceb3b96db0f3a/lua/bufferline.lua#L112-L129) and [here](https://github.com/akinsho/nvim-bufferline.lua/blob/2d633a33b055d1a8a7bf6ff6c1a7909e31778742/lua/bufferline/buffers.lua#L79-L105). However, I haven't been very successful so far, therefore i'm looking for a few pointers on how I might be able to achieve this.

Any tips would be much appreciated, cheers !

**Comment by akinsho** [comment]
@lcrockett this isn't really a use case I've designed for tbh, so I don't think it's really possible to do since the diagnostics indicator doesn't give you access to enough information to do this. Tbh the function signature at least the current arguments it passes can't be changed since that would break everyone else's configs but I guess maybe the applicable buffer could be passed in as well.

I'm not sure without looking back at the code how easy this will be to do but could just be passing in a new argument. Although at this point I'd probably make the 3rd argument a table so it has named fields and is easier to use/extend for the next time someone might want to add something

```lua
diagnostics_indicator = function(_, _, context)
   if not context.buffer:visible()
     return "" -- ideally nil should be handled so a user can just wrap the whole thing in an if statement
  end
end
```

Tbh I'm probably personally not going to implement this I understand the use case and think it's fine to support but have spread myself too thin over the last couple of months so for stuff like this I'm happy to help out re. advice on submitting a PR for this change, otherwise we can leave this open and see if someone else is happy to take it up.

**Comment by lcrockett** [comment]
Cheers for the additional context Akin. Your pointers have led me to a working example. It indeed does require exposing an additional argument to `diagnostic_indicator`. As i'm not sure what i'm doing, i'll let you judge if below seems acceptable.

```diff
--- a/lua/bufferline/diagnostics.lua
+++ b/lua/bufferline/diagnostics.lua
@@ -80,7 +80,7 @@ function M.component(context)

   local indicator = " (" .. diagnostics.count .. ")"
   if user_indicator and type(user_indicator) == "function" then
-    indicator = user_indicator(diagnostics.count, diagnostics.level, diagnostics.errors)
+    indicator = user_indicator(diagnostics.count, diagnostics.level, diagnostics.errors, context)
   end

   local highlight = highlights[diagnostics.level] or ""
```

I could then ensure the desired behaviour as follows:

```lua
    diagnostics_indicator = function(_, _, _, context)
      if context.buffer:current() then
        return ''
      end

      return ' ' .. '!'
    end,
```

Would you be willing to accept a PR for the change within `diagnostics.lua` ?

**Comment by akinsho** [comment]
Sure @lcrockett that sounds great 👍🏾 , for the actual PR though could you trim down the contents of context, I'd rather not expose too much of the internal context table to users in case people start depending on things I never intended to be exposed. Somethings `local context = { buffer = buffer}` for now and then that can grow over time if needed.

**Comment by lcrockett** [comment]
OK, I think you mean the following:

```diff
--- a/lua/bufferline/diagnostics.lua
+++ b/lua/bufferline/diagnostics.lua
@@ -80,7 +80,8 @@ function M.component(context)

   local indicator = " (" .. diagnostics.count .. ")"
   if user_indicator and type(user_indicator) == "function" then
-    indicator = user_indicator(diagnostics.count, diagnostics.level, diagnostics.errors)
+    local context = { buffer = context.buffer }
+    indicator = user_indicator(diagnostics.count, diagnostics.level, diagnostics.errors, context)
   end

   local highlight = highlights[diagnostics.level] or ""
```

If so, then this works as intended and I could leave aforementioned `diagnostics_indicator` example intact. The above example does however redefine the already existing local `context`. Is that OK or would it be better to rename the 4th argument to .. ?

**Comment by akinsho** [comment]
Yeah I'd prefer to avoid the shadowing but renaming it isn't really a big deal since what the variable is named in that function doesn't really matter since to user it's just the 4th argument, it could be `ctx` or `user_context` 🤷🏾 they can just rename it when they define their function to whatever .

It'd also be great to add this to the README and help text with your example since other users might want to do something similar.

**Comment by lcrockett** [comment]
Alrighty, while it makes sense it wouldn't matter to much to the user, i'll rename the 4th argument anyways to `ctx`, just to make LSP not mark it as a `HINT` within `diagnostics.lua`.

I'll file a PR and expand the README section `LSP Error indicators` with my example and additional information.

Cheers !

**Comment by akinsho** [comment]
Fixed by #105
